osquery/devtools/shell.cpp

```diff
@@ -1548,10 +1548,10 @@ int launchIntoShell(int argc, char** argv) {
   if (FLAGS_L || FLAGS_A.size() > 0) {
     // Helper meta commands from shell switches.
     std::string query = (FLAGS_L) ? ".tables" : ".all " + FLAGS_A;
-    char* cmd = new char[query.size() + 1];
-    memset(cmd, 0, query.size() + 1);
-    std::copy(query.begin(), query.end(), cmd);
-    rc = do_meta_command(cmd, &data);
+    boost::scoped_array<char> cmd(new char[query.size() + 1]);
+    memset(cmd.get(), 0, query.size() + 1);
+    std::copy(query.begin(), query.end(), cmd.get());
+    rc = do_meta_command(cmd.get(), &data);
```

Comment:
I agree, in this case let's opt for the `delete[] cmd` since this file is already very C89/C-extension heavy. :P